cmd/collector/app/grpc_handler.go

```diff
@@ -41,16 +40,24 @@ func NewGRPCHandler(logger *zap.Logger, spanProcessor SpanProcessor) *GRPCHandle
 
 // PostSpans implements gRPC CollectorService.
 func (g *GRPCHandler) PostSpans(ctx context.Context, r *api_v2.PostSpansRequest) (*api_v2.PostSpansResponse, error) {
-	// TODO
-	fmt.Printf("PostSpans(%+v)\n", *r)
-	for _, s := range r.Batch.Spans {
-		println(s.OperationName)
+	oks, err := g.spanProcessor.ProcessSpans(r.GetBatch().Spans, JaegerFormatType)
+	if err != nil {
+		g.logger.Error("cannot process spans", zap.Error(err))
+		return nil, err
+	}
+	success := true
+	for _, ok := range oks {
+		if !ok {
+			success = false
+			break
+		}
 	}
-	return &api_v2.PostSpansResponse{Ok: true}, nil
+	return &api_v2.PostSpansResponse{Ok: success}, nil
```

Comment:
We may want to rethink this API. What are we trying to communicate back to the caller with this Boolean flag? At least (accepted, rejected) counts would make more sense. When can the processor return not ok? Should it be returning specific errors instead?

Reply:
I agree that the API as it stands isn't exactly perfect. If anything, a boolean vector (or even bit vector) might make more sense here. Count works too if we don't care which spans failed. An optional error message might be better than the boolean response.

Reply:
Any thoughts @yurishkuro?

Reply:
The question is: what is the sender going to do with the errors?

Reply:
if we want to support retries we should be using gRPC error code for that. Can you check what openCensus is doing in this case?